### DISCUSSION TITLE
lcgeom_solver - eccentricity !=0 for contact binary

### DISCUSSION THREAD

**Discussion by kathmcp** [phoebe-project/phoebe2#503] [Questions]
Hi Kyle, sorry, I have another question!

I am following the process in the "Solvers: The Inverse Problem" tutorial, and have tried running estimator.lc_geometry (I only have light curves as observational data, and estimator.ebai wouldn't work because the eclipses are more than 0.25 phase). Although I have entered eccentricity as 0, when I try to adopt the solution from the estimator, I get the warning "WARNING contact binaries must be circular, but ecc@binary!=0  If not addressed, this warning will continue to be raised and will throw an error at run_compute." and run compute doesn't work. It is calculating a non-zero eccentricity (value 0.000245...). I have tried adjusting some input parameters, but nothing has helped. Do I need to get this to work, or can I go straight to running samplers? Or is something else wrong? My light curves already look like they are not too bad a fit (see attached). Thanks again for all your help. I am studying at home and know no-one else who uses PHOEBE, so I am so grateful for your help :)
![lightcurve](https://user-images.githubusercontent.com/81473857/126806640-3fa1474d-f35a-4ec9-ad72-8031ec6edae2.PNG)

**Comment by kecnry** [comment]
Hi Kathryn 

No worries, we're happy to help and I'm sure other people might run into these same questions.  This was actually "fixed" in the latest 2.3.50 release (#500) - none of the estimators are designed to work for contact systems and so will now raise an error stating that and not allow you to run them when having a contact configuration.  If you still want to adopt the remaining parameters for your script running an earlier version of PHOEBE, though, you can either re-set eccentricity back to zero or exclude it from `adopt_parameters` when you call `adopt_solution`.

My suggestion would be to do that, check to make sure the parameters are physical and make sense, set any other parameters manually by hand, and then move on to optimizers and eventually sampling.

-Kyle
